FlightReservation/src/screens/SignUp.js

```diff
@@ -1,118 +1,142 @@
-                            {(errors.agreeTerms && touched.agreeTerms) &&
-                            <Text>{errors.agreeTerms}</Text>}
-                            <Text>I agree with the Terms and Privacy Policy.</Text>
-                            <Text>*</Text>
-                            <CheckBox
-                                disabled={false}
-                                // onCheckColor=  '#5C6DF8'
-                                // onPress={()=> setAgreeTerms(!agreeTerms)}
-                                // value={agreeTerms}
-                            />
-                            <Text>Subscribe for select product updates.</Text>
-                        </View>
-                        <View>
-                        <Button
-                        rounded
-                        // disabled={!isValid}
-                        onPress={handleSubmit} title="Sign up" />
-                        <Text>or</Text>
-                        <Button title="Sign up with Google" />
-                        <Text>already have an account?</Text>
-                        <Text>Log in</Text>
-                        </View>
-                    </View>
-                    )}
-                </Formik>
-                </Provider>
-        </SafeAreaView>
-    );
+  const handleSignIn = values => {
+    const {name, email, password} = values;
+    setLoading(true);
+    signInWithNameEmailAndPassword(name, email, password)
+      .then(console.log)
+      .catch(console.log)
+      .finally(() => setLoading(false));
+  };
+
+  return (
+    <SafeAreaView>
+      <Provider store={store}>
+        {loading && <Text>Loading...</Text>}
+        <Formik
+          validationSchema={signUpValidationSchema}
+          initialValues={{
+            name: 'Juan',
+            email: 'juan1@example.com',
+            password: 'Juan123%',
+            agreeTerms: true,
+          }}
+          validateOnMount={true}
+          onSubmit={values => handleSignIn(values)}>
+          {({
+            handleChange,
+            handleBlur,
+            handleSubmit,
+            setFieldValue,
+            values,
+            touched,
+            errors,
+            isValid,
+          }) => (
+            <View>
+              <Text>Sign up</Text>
+              <View>
+                <Text>Name</Text>
+                <TextInput
+                  onChangeText={handleChange('name')}
+                  onBlur={handleBlur('name')}
+                  value={values.name}
+                />
+                {errors.name && touched.name && <Text>{errors.name}</Text>}
+              </View>
+              <View>
+                <Text>Email</Text>
+                <TextInput
+                  onChangeText={handleChange('email')}
+                  onBlur={handleBlur('email')}
+                  value={values.email}
+                />
+                {errors.email && touched.email && <Text>{errors.email}</Text>}
+              </View>
```

Comment:
Most of the inputs, title, and error text could be extracted as a component, tools of code improvement like CodeClimate will detect this as code repeated.
The rest of the code looks great.